components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -126,7 +128,20 @@ public AbstractAPIManager(String username) throws APIManagementException {
                 //load resources for each tenants.
                 APIUtil.loadloadTenantAPIRXT( tenantUserName, tenantId);
                 APIUtil.loadTenantAPIPolicy( tenantUserName, tenantId);
-                APIUtil.writeDefinedSequencesToTenantRegistry(tenantId);
+
+                //Check whether GatewayType is synapse before loading sequences into registry
+                String filePath =
+                        CarbonUtils.getCarbonHome() + File.separator + "repository" + File.separator + "conf" +
+                                File.separator + "api-manager.xml";
+
+                APIManagerConfiguration configuration = new APIManagerConfiguration();
+                configuration.load(filePath);
```

Comment:
You can use the APIManagerConfigurationService OSGI service available with out reading api-manager.xml again here. 

APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().
                getAPIManagerConfiguration();